### ISSUE TITLE
The insert operation must provide keyColumn, but no need  before

### ISSUE THREAD

**Issue by snyang** [chenggangpro/reactive-mybatis-support#53]
The following content worked when use mybatis, but failed when use mybatis_r2dbc,

```xml
<insert id="insert" parameterType="SysUser" useGeneratedKeys="true" keyProperty="userId">
```

To get the mybatis_r2dbc working, I have to provide values for keyColumn elements
```xml
<insert id="insert" parameterType="SysUser" useGeneratedKeys="true" keyProperty="userId" keyColumn="user_id">
```

**Comment by chenggangpro** [comment]
Could your provide a specific description , or demo project？

   I checked my own used case and test suite about that,It works properly. [Test case about insert return generated keys](https://github.com/chenggangpro/reactive-mybatis-support/blob/13c20cd0d9a019a2c55bdbf5f8985ae8587020ee/mybatis-r2dbc/src/test/java/pro/chenggang/project/reactive/mybatis/support/r2dbc/suite/SimpleMapperTests.java#L89)
> FYI , The test running with `mysql` and `mariadb-driver`

**Comment by chenggangpro** [comment]
Sorry, I misunderstood, I will try to fix the problem later

**Comment by chenggangpro** [comment]
Hi~,
    I have checked the source code ,the problem you mentioned  is actually caused by the inconsistency between `JDBC driver` and `R2DBC driver` definition.
* 1 . the `JDBC driver`'s`Connection` provides a `PreparedStatement prepareStatement(String sql, int autoGeneratedKeys) throws SQLException;` method, the second parameter of this method indicates whether auto returns generatedKeys
* 2 . The `R2DBC driver`'s `Statement` does not provide any method to return generatedKeys automatically, the Statement only has a `Statement returnGeneratedValues(String... columns)` method. columns)` method to return generatedKeys, so there is no bug here.

Reference:

* 1 . [JDBC's Connection](https://docs.oracle.com/javase/8/docs/api/java/sql/Connection.html#prepareStatement-java.lang.String-int-)
* 2 . [R2DBC's Statement](https://github.com/r2dbc/r2dbc-spi/blob/8e27a1df8caf4688942bef1fa60848967586c336/r2dbc-spi/src/main/java/io/r2dbc/spi/Statement.java#L115)

**Comment by snyang** [comment]
Thanks. better to throw an exception for the case.

**Comment by chenggangpro** [comment]
> Thanks. better to throw an exception for the case.

Sounds appropriate.I'll work on this.Thanks for your suggestion

**Comment by chenggangpro** [comment]
FYI: fixed in #56
